pkg/controller/restore-operator/http.go

```diff
@@ -73,12 +73,18 @@ func (r *Restore) serveBackup(w http.ResponseWriter, req *http.Request) error {
 
 	logrus.Infof("serving backup for restore CR %v", restoreName)
 	cr := v.(*api.EtcdRestore)
-	restoreSource := cr.Spec.RestoreSource
-	var backupReader reader.Reader
-	var path string
 
-	switch {
-	case restoreSource.S3 != nil:
+	var (
+		backupReader reader.Reader
+		path         string
+	)
+
+	switch cr.Spec.BackupStorageType {
+	case api.BackupStorageTypeS3:
+		restoreSource := cr.Spec.RestoreSource
+		if restoreSource.S3 == nil {
+			return errors.New("empty s3 restore source")
+		}
```

Comment:
```
default:
  return ("unknown backup storage type (%s) for restore CR (%s)
```

Reply:
@hongchaodeng do we want to print everything about restore CR? that can be very verbose.

Reply:
just the name

Reply:
kk
i have this
`return fmt.Errorf("unknown backup storage type (%s) for restore-name (%v)", cr.Spec.BackupStorageType, restoreName)`

Reply:
restore-name -> restore CR

Reply:
@hongchaodeng  it should be `restore-name` to be consistent. code uses `restore-name` in other places too.

Reply:
That's based on different contexts:
```
		return fmt.Errorf("failed to get restore CR for restore-name (%v): %v", restoreName, err)
...
		return fmt.Errorf("no restore CR found for restore-name (%v)", restoreName)
```